compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2393,6 +2359,164 @@ public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
+        }
+        return null;
+    }
+
+    private boolean isDifferentPackage(BType type) {
+        return type.tsymbol.owner.tag == SymTag.PACKAGE && !type.tsymbol.pkgID.equals(env.enclPkg.packageID);
+    }
+
+    private BLangPackageVarRef createPackageVarRef(BSymbol symbol) {
+        BLangPackageVarRef packageVarRef = new BLangPackageVarRef((BVarSymbol) symbol);
+        packageVarRef.pos = symbol.pos;
+        packageVarRef.setBType(symbol.getType());
+        return packageVarRef;
+    }
+
+    private BIRVariableDcl findInLocalSymbolVarMap(BType type, Map<BSymbol, BIRVariableDcl> varMap) {
+        for (Map.Entry<BSymbol, BIRVariableDcl> entry : varMap.entrySet()) {
+            if (isTypeDescSymbol(entry.getKey(), Types.getImpliedType(type))) {
+                return varMap.get(entry.getKey());
+            }
+        }
+        return null;
+    }
+
+    private BIRVariableDcl findInGlobalSymbolVarMap(BType type, Map<BSymbol, BIRGlobalVariableDcl> varMap) {
+        for (Map.Entry<BSymbol, BIRGlobalVariableDcl> entry : varMap.entrySet()) {
+            if (isTypeDescSymbol(entry.getKey(), Types.getImpliedType(type))) {
+                return varMap.get(entry.getKey());
+            }
+        }
+        return null;
+    }
+
+    private boolean isTypeDescSymbol(BSymbol symbol, BType targetType) {
+        return symbol.type.tag == TypeTags.TYPEDESC && ((BTypedescType) symbol.type).constraint == targetType;
+    }
+
+    private void createNewTypedescInst(Location position, BType type, BType resolveType) {
+        BTypeSymbol typeSymbol = resolveType.tsymbol;
+        BIRVariableDcl tempVarDcl = createTempVariable(type);
+        BIROperand toVarRef = new BIROperand(tempVarDcl);
+
+        BIRNonTerminator.NewTypeDesc newTypeDesc = createNewTypeDesc(position, toVarRef, resolveType, typeSymbol);
+
+        this.env.targetOperand = toVarRef;
+        setScopeAndEmit(newTypeDesc);
+    }
+
+    private BIRVariableDcl createTempVariable(BType type) {
+        BIRVariableDcl tempVarDcl = new BIRVariableDcl(type, this.env.nextLocalVarId(names), VarScope.FUNCTION,
+                                                       VarKind.TEMP);
+        this.env.enclFunc.localVars.add(tempVarDcl);
+        return tempVarDcl;
+    }
+
+    private BIRNonTerminator.NewTypeDesc createNewTypeDesc(Location position, BIROperand toVarRef, BType resolveType,
+                                                           BTypeSymbol typeSymbol) {
+        List<BIROperand> closures = Collections.emptyList();
+        if (typeSymbol != null && typeSymbol.annotations != null) {
+            BIROperand symbolVarOperand = this.env.symbolVarMap.containsKey(typeSymbol.annotations) ?
+                    new BIROperand(this.env.symbolVarMap.get(typeSymbol.annotations)) :
+                    new BIROperand(this.globalVarMap.get(typeSymbol.annotations));
+            return new BIRNonTerminator.NewTypeDesc(position, toVarRef, resolveType, closures, symbolVarOperand,
```

Comment:
```suggestion
            return new BIRNonTerminator.NewTypeDesc(position, toVarRef, resolveType, Collections.emptyList(), symbolVarOperand,
```
Its not necessary to keep a variable.